include/bluetooth/gatt.h

```diff
@@ -677,6 +677,36 @@ ssize_t bt_gatt_attr_read_cpf(struct bt_conn *conn,
 	.user_data = _value,						\
 }
 
+/** @brief GATT callback structure.
+ *
+ *  It is registered with the help of the bt_gatt_cb_register() API.
+ *  It's permissible to register multiple instances of this @ref bt_gatt_cb_register
+ *  type, in case different modules of an application are interested in
+ *  tracking the GATT request state. If a callback is not of interest for
+ *  an instance, it may be set to NULL and will as a consequence not be
+ *  used for that instance.
+ */
+struct bt_gatt_notify_cb {
+	/** @brief The notification TX completed.
+	 *
+	 *  This callback informs the application
+	 *  that a notification was transmitted.
+	 *
+	 *  @param conn New connection object.
+	 */
+	void (*notify_complete)(struct bt_conn *conn,
+				const struct bt_gatt_attr *attr);
+	struct bt_gatt_notify_cb *_next;
```

Comment:
Can we make this the first member in struct (slist node instead) and use slist interfaces for maintaining?